x-pack/plugins/lens/public/editor_frame_service/editor_frame/config_panel/chart_switch/chart_switch.tsx

```diff
@@ -483,6 +485,7 @@ function getTopSuggestion(
 
   return (
     suggestions.find((s) => s.changeType === 'unchanged' || s.changeType === 'reduced') ||
+    suggestions.find((s) => s.keptLayerIds.some((id) => id === layerId)) ||
     suggestions[0]
```

Comment:
actually no, because we want to give a bigger priority to unchanged or reduced types. And only if we cannot find one of these, we'll be searching through the rest (changeType layers is what matters here)